tests/sles4sap/sapconf.pm

```diff
@@ -128,9 +129,10 @@ sub run {
     verify_sapconf_service('sysstat.service', 'Write information about system start to sysstat log')
       if (is_sle('>=15'));
 
-    my $statusregex = join('.+', @tuned_profiles);
     $output = script_output "tuned-adm list";
-    die "Command 'tuned-adm list' output is not recognized" unless ($output =~ m|$statusregex|s);
+    foreach my $p (@tuned_profiles) {
+        die "Profile $p is not in 'tuned-adm list'" unless (grep(/$p/, $output));
```

Comment:
I would've gone with `unless ($output =~ m|$p|)` instead of the `grep`, but I guess the `grep` is more readable.

WRT the loop vs. the regexp, I think they basically check for 2 different things: the regexp was checking that the output to `tuned-adm list` was the expected one, so indeed, order was important. The loop instead checks for the presence of each profile name within the output, which given some of the current profile names and future changes to the output could lead to false positives IMO. For example, given the output from the verification run:

```
Available profiles:
- balanced                    - General non-specialized tuned profile
- cpu-partitioning            - Optimize for CPU partitioning
- desktop                     - Optimize for the desktop use-case
- latency-performance         - Optimize for deterministic performance at the cost of increased power consumption
- mssql                       - Optimize for MS SQL Server
- network-latency             - Optimize for deterministic performance at the cost of increased power consumption, focused on low latency network performance
- network-throughput          - Optimize for streaming network throughput, generally only necessary on older CPUs or 40G+ networks
- powersave                   - Optimize for low power consumption
- sapconf                     - Optimize for SAP workloads like NetWeaver, SAP HANA and HANA based products
- saptune                     - Comprehensive system optimisation with saptune
- throughput-performance      - Broadly applicable tuning that provides excellent performance across a variety of common server workloads
- virtual-guest               - Optimize for running inside a virtual guest
- virtual-host                - Optimize for running KVM guests
Current active profile: saptune
```

Profiles with names that are too generic (for example `desktop` or `balanced`) can be a problem, although in this case they're OK. Also, `saptune` in that specific example is matching 2 lines, which again, means the profile is available, but if the intention was to match the text only on the list, then it's a possible false positive. Imagine the active profile was `saptune` but the profile itself was not shown in the list as an available profile ... is that a bug, or not?

> But I think it's better to test each profile in a loop (can be a little bit slower but not that much) and this way if a profile name position is changed we don't have to change the code, WDYT?

This is being done between L142 and L147 by calling `tuned-adm profile $p`.

Reply:
> WRT the loop vs. the regexp, I think they basically check for 2 different things: the regexp was checking that the output to tuned-adm list was the expected one, so indeed, order was important.

I don't understand the purpose of this test: is anyone asked to check the order? It seems to be alphabetical order but this can changed and it's not really important IMHO, again apart if this order have to be explicitly checked.

Anyway I think that I can "simply" sort the value in the array and then re-use the regexp for this test.

> Also, `saptune` in that specific example is matching 2 lines, which again, means the profile is available, but if the intention was to match the text only on the list, then it's a possible false positive. Imagine the active profile was `saptune` but the profile itself was not shown in the list as an available profile ... is that a bug, or not?

Yes, I could do a grep on `- ${profile}`, better in that specific case. But as I said above I think that I will re-use the regexp.

Reply:
> I don't understand the purpose of this test: is anyone asked to check the order?

Not as far as I know. IIRC, the original test was written to test all available options/arguments to the `sapconf` command, so when this was changed due to the new `sapconf`, it was adapted to check all available options/arguments to `tuned-adm`. This includes `tuned-adm list`, but I guess we can drop that if it's causing more problems than the value it provides, after all, we will be checking all expected profiles in the loop within L142 and L147 and with the change this PR introduces we will only be checking `virtual-host` when backend is not qemu. We may need to keep up to date when new profiles are added or removed though.

> Anyway I think that I can "simply" sort the value in the array and then re-use the regexp for this test.

Yes, either that or remove the check for the output of `tuned-adm list`. Whichever you think it's better.